data-processing-lib/ray/src/data_processing_ray/runtime/ray/transform_file_processor.py

```diff
@@ -31,16 +31,14 @@ def __init__(self, params: dict[str, Any]):
             transform_params: dictionary of parameters for local transform creation
             statistics: object reference to statistics
         """
-        super().__init__()
-        # Create data access
-        self.data_access = params.get("data_access_factory", None).create_data_access()
-        # Add data access ant statistics to the processor parameters
-        transform_params = params.get("transform_params", None)
-        transform_params["data_access"] = self.data_access
+        super().__init__(
+            data_access_factory=params.get("data_access_factory", None),
+            transform_parameters=dict(params.get("transform_params", {})),
+        )
+        self.transform_params["statistics"] = params.get("statistics", None)
```

Comment:
Should we be better about handling None?

Reply:
We do

Reply:
Not sure I was clear.  If "statistics" is not provided, then self.stats is None, but then later it is used in _publish_stats() and this will cause an error.

Reply:
Its fixed

Reply:
"statistics" is fixed, but there are lots of other places not checking for None.  But, that said, let's fix this as we go. I won't hold up this one over that